src/coreclr/jit/lsraloongarch64.cpp

```diff
@@ -0,0 +1,1377 @@
+            }
+            FALLTHROUGH;
+
+        case GT_AND:
+        case GT_OR:
+        case GT_XOR:
+        case GT_LSH:
+        case GT_RSH:
+        case GT_RSZ:
+        case GT_ROR:
+            srcCount = BuildBinaryUses(tree->AsOp());
+            buildInternalRegisterUses();
+            assert(dstCount == 1);
+            BuildDef(tree);
+            break;
+
+        case GT_RETURNTRAP:
+            // this just turns into a compare of its child with an int
+            // + a conditional call
+            BuildUse(tree->gtGetOp1());
+            srcCount = 1;
+            assert(dstCount == 0);
+            killMask = compiler->compHelperCallKillSet(CORINFO_HELP_STOP_FOR_GC);
+            BuildDefsWithKills(tree, 0, RBM_NONE, killMask);
+            break;
+
+        case GT_MUL:
+        case GT_MOD:
+        case GT_UMOD:
+        case GT_DIV:
+        case GT_MULHI:
+        case GT_UDIV:
+        {
+            srcCount = BuildBinaryUses(tree->AsOp());
+            buildInternalRegisterUses();
+            assert(dstCount == 1);
+            BuildDef(tree);
+        }
+        break;
+
+        case GT_INTRINSIC:
+        {
+            noway_assert((tree->AsIntrinsic()->gtIntrinsicName == NI_System_Math_Abs) ||
+                         (tree->AsIntrinsic()->gtIntrinsicName == NI_System_Math_Ceiling) ||
+                         (tree->AsIntrinsic()->gtIntrinsicName == NI_System_Math_Floor) ||
+                         (tree->AsIntrinsic()->gtIntrinsicName == NI_System_Math_Round) ||
+                         (tree->AsIntrinsic()->gtIntrinsicName == NI_System_Math_Sqrt));
+
+            // Both operand and its result must be of the same floating point type.
+            GenTree* op1 = tree->gtGetOp1();
+            assert(varTypeIsFloating(op1));
+            assert(op1->TypeGet() == tree->TypeGet());
+
+            BuildUse(op1);
+            srcCount = 1;
+            assert(dstCount == 1);
+            BuildDef(tree);
+        }
+        break;
+
+#ifdef FEATURE_SIMD
+        case GT_SIMD:
+            srcCount = BuildSIMD(tree->AsSIMD());
+            break;
+#endif // FEATURE_SIMD
+
+#ifdef FEATURE_HW_INTRINSICS
+        case GT_HWINTRINSIC:
+            srcCount = BuildHWIntrinsic(tree->AsHWIntrinsic(), &dstCount);
+            break;
+#endif // FEATURE_HW_INTRINSICS
+
+        case GT_CAST:
+            assert(dstCount == 1);
+            srcCount = BuildCast(tree->AsCast());
+            break;
+
+        case GT_NEG:
+        case GT_NOT:
+            BuildUse(tree->gtGetOp1());
+            srcCount = 1;
+            assert(dstCount == 1);
+            BuildDef(tree);
+            break;
+
+        case GT_EQ:
+        case GT_NE:
+        case GT_LT:
+        case GT_LE:
+        case GT_GE:
+        case GT_GT:
+        case GT_JCMP:
+            if (!varTypeIsFloating(tree->gtGetOp1()))
+            {
+                // We need two registers: tmpRegOp1 and tmpRegOp2
```

Comment:
I missed this earlier, why is there a need of two registers here?

Reply:
OK, I will delete it.
In future we will refactor this for LoongArch64.

Thanks.

Reply:
The LA ISA about compare or branch,  especially for the 32-bits and unsigned type, maybe used the temp registers for comparing.